src/detect-engine-alert.c

```diff
@@ -183,15 +188,17 @@ static void PacketApplySignatureActions(Packet *p, const Signature *s, const uin
     SCLogDebug("packet %" PRIu64 " sid %u action %02x alert_flags %02x", p->pcap_cnt, s->id,
             s->action, alert_flags);
 
-    if (s->action & ACTION_DROP) {
-        PacketDrop(p, PKT_DROP_REASON_RULES);
+    if (s->action & (ACTION_DROP | ACTION_REJECT_ANY)) {
```

Comment:
looking at how `s->action` is set up, this is a bit redundant. For reject we always also set `ACTION_DROP`. There is no harm in showing the reject here, but I would like 2 additions:
1. a comment stating this
2. a `DEBUG_VALIDATE_BUG_ON` that makes sure `ACTION_DROP` is always set in this branch